packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -0,0 +1,340 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as eth from "./ethereum";
+import _ from "lodash";
+
+class XDeFiTransport extends core.Transport {
+  public async getDeviceID() {
+    return "xdefi:0";
+  }
+
+  public call(...args: any[]): Promise<any> {
+    return Promise.resolve();
+  }
+}
+
+export function isXDeFi(wallet: core.HDWallet): wallet is XDeFiHDWallet {
+  return _.isObject(wallet) && (wallet as any).__XDEFI;
+}
+
+type HasNonTrivialConstructor<T> = T extends { new (): any } ? never : T;
+
+export class XDeFiHDWallet implements core.HDWallet, core.ETHWallet {
+  readonly _supportsETH = true;
+  readonly _supportsETHInfo = true;
+  readonly _supportsBTCInfo = false;
+  readonly _supportsBTC = false;
+  readonly _supportsCosmosInfo = false;
+  readonly _supportsCosmos = false;
+  readonly _supportsOsmosisInfo = false;
+  readonly _supportsOsmosis = false;
+  readonly _supportsBinanceInfo = false;
+  readonly _supportsBinance = false;
+  readonly _supportsDebugLink = false;
+  readonly _isPortis = false;
+  readonly _isMetaMask = false;
+  readonly __XDEFI = true;
+  readonly _supportsRippleInfo = false;
+  readonly _supportsRipple = false;
+  readonly _supportsEosInfo = false;
+  readonly _supportsEos = false;
+  readonly _supportsFioInfo = false;
+  readonly _supportsFio = false;
+  readonly _supportsThorchainInfo = false;
+  readonly _supportsThorchain = false;
+  readonly _supportsSecretInfo = false;
+  readonly _supportsSecret = false;
+  readonly _supportsKava = false;
+  readonly _supportsKavaInfo = false;
+  readonly _supportsTerra = false;
+  readonly _supportsTerraInfo = false;
```

Comment:
- nothing good will come of breaking the naming convention
> _alles muss in ordnung kommen_
- all the `= false` can die
```suggestion
  readonly _supportsETH = true;
  readonly _isXDeFi = true;
```